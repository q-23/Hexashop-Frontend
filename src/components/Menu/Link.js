import { Link as RouterLink } from 'react-router-dom';
import styled from 'styled-components';
import * as palette from '../../assets/css_variables/colors';

const Link = styled(RouterLink)`
		${({activelink}) => activelink ? `
			text-decoration: underline;
    	color: white;
		` : `
			text-decoration: none;
		`};
		font-family: 'Lato', sans-serif;
    text-decoration-color: ${palette.LIGHT_PURPLE};
    text-transform: uppercase;
    font-size: 1.2em;
    color: white;
    font-weight: 700;
`;

export default Link;