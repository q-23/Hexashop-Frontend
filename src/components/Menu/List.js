import styled from 'styled-components';

const List = styled.ul`
		text-align: right;
		margin: 10px 0;
		width: 7.5em;
		padding: 0;
`;

export default List;