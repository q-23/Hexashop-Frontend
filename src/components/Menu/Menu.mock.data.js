export const MOCK_CATEGORIES = [
	{to: '/', label: 'Start'},
	{to: '/shoes', label: 'Shoes'},
	{to: '/jackets', label: 'Jackets'},
	{to: '/pants', label: 'Pants'},
	{to: '/caps', label: 'Caps'}
]
