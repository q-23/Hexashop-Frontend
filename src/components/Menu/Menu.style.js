import * as palette from 'assets/css_variables/colors';
import { Link as RouterLink } from 'react-router-dom';
import WithWidth from "components/_HOC/WithWidth";

import styled from "styled-components";

const Wrapper = styled.nav`
  height: 100vh;
  background: ${palette.BLUE};
  padding: 3.5em 2em 0 0;
  position: fixed;
  top: 0;
  left: 0;
  z-index: 90;
  transition: transform .3s cubic-bezier(0, .52, 0, 1);
  overflow: scroll;
	transform: translate3d(-100vw , 0, 0);
	box-shadow: ${({ visible }) => visible ? '10px 2px 23px -1px rgba(0,0,0,0.62)' : 'none'};
	${({ visible }) => visible && `
	  transform: translate3d(0vw, 0, 0);
	  overflow: auto;
		::-webkit-scrollbar {
			width: 10px;
		}
		::-webkit-scrollbar-track {
			background: transparent; 
		}
		::-webkit-scrollbar-thumb {
			background: #2d46bf; 
		}
		::-webkit-scrollbar-thumb:hover {
			background: #1f37af; 
		}
	`};
`;

const MenuOverlay = styled.div`
	backdrop-filter: blur(5px);
  visibility: hidden;
  overflow: hidden;
  position: fixed;
  height: 100vh;
	width: 100vw;
  z-index: 89;
  left: 0;
  top: 0;
  
	${({ visible }) => visible && `
	visibility: visible;
	transform: translate3d(0vw, 0, 0);
	`};
`;

const List = styled.ul`
		padding-left: 20px;
		list-style: none;
		width: 100%;
		height: 95%;
`;

const ListItem = styled.li`
	margin: 10px 0;
	width: 100%;
`


const Link = styled(RouterLink)`
		${({activelink}) => activelink ? `
			text-decoration: underline;
    	color: white;
		` : `
			text-decoration: none;
		`};
    text-decoration-color: ${palette.LIGHT_PURPLE};
		font-family: 'Lato', sans-serif;
    text-transform: uppercase;
    font-size: 1.2em;
    font-weight: 700;
		color: white;
`;


const MenuWrapper = WithWidth(Wrapper);

export {
	MenuWrapper,
	MenuOverlay,
	ListItem,
	Link,
	List
}