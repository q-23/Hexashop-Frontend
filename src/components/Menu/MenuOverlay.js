import styled from "styled-components";

const MenuOverlay = styled.div`
  width: 100vw;
  height: 100vh;
	backdrop-filter: blur(5px);
  position: fixed;
  top: 0;
  left: 0;
  z-index: 89;
  overflow: hidden;
  visibility: hidden;
  
	${({ visible }) => visible && `
	visibility: visible;
	  transform: translate3d(0vw, 0, 0);
	`};
`;

export default MenuOverlay;