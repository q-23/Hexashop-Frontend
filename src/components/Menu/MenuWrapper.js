import styled from "styled-components";
import * as palette from '../../assets/css_variables/colors';

const MenuWrapper = styled.nav`
  width: 25vw;
  height: 100vh;
  background: ${palette.BLUE};
  position: fixed;
  top: 0;
  left: 0;
  z-index: 90;
  transition: transform .3s cubic-bezier(0, .52, 0, 1);
  overflow: scroll;
  padding-top: 3.5em;
  transform: translate3d(-26vw, 0, 0);
	box-shadow: 10px 2px 23px -1px rgba(0,0,0,0.62);
	${({ visible }) => visible && `
	  transform: translate3d(0vw, 0, 0);
	  overflow: auto;
		::-webkit-scrollbar {
			width: 10px;
		}
		::-webkit-scrollbar-track {
			background: transparent; 
		}
		::-webkit-scrollbar-thumb {
			background: #2d46bf; 
		}
		::-webkit-scrollbar-thumb:hover {
			background: #1f37af; 
		}
	`};
`;

export default MenuWrapper;