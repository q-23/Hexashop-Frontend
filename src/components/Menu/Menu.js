import React, { useContext } from "react";

import { withRouter } from 'react-router-dom';

import { MenuContext } from 'contexts/menu/menu';

import {MenuOverlay, MenuWrapper, Link, List} from './Menu.style'

const Menu = ({ location }) => {
	const { menuOpen, setMenuOpen } = useContext(MenuContext);
	const { menuCategories } = useContext(MenuContext);

	const { pathname } = location;

	return(
		<>
			<MenuWrapper visible={menuOpen}>
				<ul>
					{
						menuCategories
							.map(el =>
							<List key={`${el.category_path} - ${el.category_name}`}>
								<Link
									to={`${el.category_path}`}
									activelink={pathname === el.category_path ? 1 : 0}
									onClick={() => {
										setMenuOpen(false)
									}}
								>
									{el.category_name}
								</Link>
							</List>)
					}
				</ul>
			</MenuWrapper>
			<MenuOverlay visible={menuOpen} onClick={() => setMenuOpen(false)}/>
		</>
	)
};

export default withRouter(Menu);