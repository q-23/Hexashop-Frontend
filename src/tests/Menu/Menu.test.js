import React from 'react';

import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/react'
import {renderWithRouter} from "../utils";

import Menu from '../../components/Menu';

import { MOCK_CATEGORIES } from "./Menu.mock.data";

describe('[MENU]', () => {
	test('Should render without failing with empty data', () => {
		const wrapper = render(
			renderWithRouter(<Menu />)
		);
		expect(wrapper).toMatchSnapshot();
	})

	test('Should not render list without data given', () => {
		const wrapper = render(renderWithRouter(<Menu categories={[]}/>));
		const { container } = wrapper;
		expect(wrapper).toMatchSnapshot();
		expect(container.querySelector('li')).toBeFalsy()
	});

	test('Should render lists of links', () => {
		const wrapper = render(renderWithRouter(<Menu categories={MOCK_CATEGORIES}/>));
		const { getByText } = wrapper;
		MOCK_CATEGORIES.forEach(link => {
			expect(getByText(link.label)).toBeInTheDocument()
			expect(getByText(link.label)).toHaveAttribute('href', link.to)
		})
		expect(wrapper).toMatchSnapshot();
	});

	test('Should add underline to active links', () => {
		const wrapper = render(renderWithRouter(<Menu categories={MOCK_CATEGORIES}/>, '/shoes'));
		const { getByText } = wrapper;
		expect(getByText('Shoes')).toHaveStyle(`text-decoration: underline;`)
	});

})

