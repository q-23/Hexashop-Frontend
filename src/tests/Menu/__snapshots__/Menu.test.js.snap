// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[MENU] Should not render list without data given 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="sc-AxjAm kdbTIa"
        width="md"
      >
        <ul />
      </nav>
      <div
        class="sc-AxirZ fYEXoq"
      />
    </div>
  </body>,
  "container": <div>
    <nav
      class="sc-AxjAm kdbTIa"
      width="md"
    >
      <ul />
    </nav>
    <div
      class="sc-AxirZ fYEXoq"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`[MENU] Should render lists of links 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="sc-AxjAm kdbTIa"
        width="md"
      >
        <ul>
          <ul
            class="sc-AxiKw gSZvgP"
          >
            <a
              activelink="0"
              class="sc-AxhCb inCRGH"
              href="/cat1"
            >
              First category
            </a>
          </ul>
        </ul>
      </nav>
      <div
        class="sc-AxirZ fYEXoq"
      />
    </div>
  </body>,
  "container": <div>
    <nav
      class="sc-AxjAm kdbTIa"
      width="md"
    >
      <ul>
        <ul
          class="sc-AxiKw gSZvgP"
        >
          <a
            activelink="0"
            class="sc-AxhCb inCRGH"
            href="/cat1"
          >
            First category
          </a>
        </ul>
      </ul>
    </nav>
    <div
      class="sc-AxirZ fYEXoq"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`[MENU] Should render without failing with empty data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="sc-AxjAm kdbTIa"
        width="md"
      >
        <ul />
      </nav>
      <div
        class="sc-AxirZ fYEXoq"
      />
    </div>
  </body>,
  "container": <div>
    <nav
      class="sc-AxjAm kdbTIa"
      width="md"
    >
      <ul />
    </nav>
    <div
      class="sc-AxirZ fYEXoq"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
